kyo-data/shared/src/main/scala/kyo/internal/SafeClassTagMacro.scala

```diff
@@ -50,45 +50,50 @@ private[kyo] object SafeClassTagMacro:
                             case _             => Chunk(tpe)
                     val exprs = flatten(tpe).map(create)
                     '{ Intersection(Set(${ Varargs(exprs) }*)) }
-                case _ => createSingle(tpe)
+                case _ => createSingle(tpe).asInstanceOf[Expr[ConcreteTag[Any]]]
             end match
         end create
 
         def createSingle(tpe: TypeRepr): Expr[ConcreteTag[Any]] =
-            checkType(tpe)
-            tpe match
-                case ConstantType(const) =>
-                    val value =
-                        const.value match
-                            case x: Int     => Expr(x)
-                            case x: Long    => Expr(x)
-                            case x: Float   => Expr(x)
-                            case x: Double  => Expr(x)
-                            case x: Boolean => Expr(x)
-                            case x: Char    => Expr(x)
-                            case x: String  => Expr(x)
-                            case x          => report.errorAndAbort(s"Unsupported literal type: $x")
-                    '{ LiteralTag($value) }
+            Implicits.search(TypeRepr.of[[A] =>> ConcreteTag[A]].appliedTo(tpe)) match
```

Comment:
The macro change looks ok to me, it should have been this way from the beginning. What do you think about separating this change in a new PR and adding tests for it?